pkg/kubernetes/nulecule.js

```diff
@@ -181,30 +184,96 @@ define([
-                .done(function(data) {
-                    deferred.resolve();
-                })
+            var docker_service = service.proxy("docker");
+            docker_service.start()
                 .fail(function(ex) {
-                    var tag = "latest";
-                    var tagl = image.split(":");
-                    if(tagl.length > 1)
-                        tag = tag[1];
-
-                    docker.pull(image, tag)
-                        .progress(function(msg) {
-                            deferred.notify(msg);
-                        })
-                        .done(function(d) {
+                    console.warn(ex);
+                    var message = _("Unable to start docker");
+                    var err = new Error(message);
+                    deferred.reject(err);
+                })
+                .done(function(data) {
+                    docker.inspect_image(image)
+                        .done(function(data) {
                             deferred.resolve();
                         })
                         .fail(function(ex) {
-                            deferred.reject(ex);
+                            var tag = "latest";
+                            var tagl = image.split(":");
+                            if(tagl.length > 1)
+                                tag = tag[1];
+
+                            docker.pull(image, tag)
+                                .progress(function(msg) {
+                                    deferred.notify(msg);
+                                })
+                                .done(function(d) {
+                                    deferred.resolve();
+                                })
+                                .fail(function(ex) {
+                                    deferred.reject(ex);
+                                });
                         });
                 });
             return deferred.promise();
         }
 
+        function is_flavor_present(providers, flavor) {
+            if (providers && providers.indexOf(flavor) >= 0) {
+                return true;
+            }
+            return false;
+        }
+
+        function format_provider_list(providers) {
+            var providersl = providers.split(",");
+            var newproviders = [];
+            for (var i = providersl.length - 1; i >= 0; i--) {
+                newproviders.push(providersl[i].trim());
+            }
+            return newproviders;
+        }
+
+        function check_flavor_supported(versions) {
+            var providersl = format_provider_list(versions.providers);
+            var supported = false;
+            if (client.flavor === "openshift" || client.flavor === "kubernetes") {
+                supported = is_flavor_present(providersl, client.flavor);
+            } else {
+                supported = false;
+            }
+            return supported;
+        }
+
+        function versionCompare(v1, v2) {
```

Comment:
This should be broken out, and unit tests should be added for it. We could eventually add it as an API to cockpit.js and add documentation, but for now putting it in kuberenetes/config.js and adding unit tests should be good enough.